htdocs/mrp/mo_production.php

```diff
@@ -149,7 +151,7 @@
 		$moline->fk_mo = $object->id;
 		$moline->qty = GETPOST('qtytoadd', 'int'); ;
 		$moline->fk_product = GETPOST('productidtoadd', 'int');
-		$moline->role = 'toconsume';
+		$moline->role = 'toconsumef'; // free consume line
```

Comment:
Why did you changed key "toconsume"  into  "toconsumef"
A lot of other code is using "toconsume" to calculate virtual stock so key is important.

Reply:
I changed to see when line comes from bom or from 'addconsumeline' to add also lines cost from 'addconsumeline' to production unitcost.
First PR was on 12 and wanted to avoid database change, but now for develop branch maybe add new field in llx_mrp_production for this or maybe you have better idea to solve this?
Also not really understand this feature 'addconsumeline', why change bom after manufacturing order started?

Reply:
Yes adding new fields is better. For example, you can add the 2 fields
origin_id (id of bom or empty) + origin_type ('bom', 'free', ...)

Reply:
Ok will add new fields. Will also check 'addconsumeline' again, I removed add line when production started, after your explication better only remove when production closed.

Reply:
Add new fields and allow add free line after production started.
New we have origin id, I replaced show consumed pmp by consumed unitcost.